search.py

```diff
@@ -181,7 +181,7 @@ def tree_search(problem, frontier):
     while frontier:
         node = frontier.pop()
         if problem.goal_test(node.state):
-            return node
+            yield node
```

Comment:
No, you can't change the `search` algorithms to be generators instead of functions. You'll need to define a separate function if you want to yield all possible solutions.

Reply:
Thanks for the tip. I will revert this in my next push.